vendor/github.com/containers/common/libimage/manifests/manifests.go

```diff
@@ -239,6 +241,16 @@ func (l *list) Push(ctx context.Context, dest types.ImageReference, options Push
 			return nil, "", err
 		}
 	}
+
+	compressionVariants := []cp.CopyOptionCompressionVariant{}
+	//options.EnsureCompressionVariantExists = []CopyOptionCompressionVariant{CopyOptionCompressionVariant{Algorithm: &compression.Zstd, Level:20}}
+	for _, compressionVariant := range options.ReplicateWithCompression {
+		if compressionVariant == "zstd" {
```

Comment:
Else this should not silently do _nothing_; with code as is, any future feature addition could break a working user.

Instead, any unexpected value should cause a failure.

Reply:
Yes I expect lot of changes in here, everything in this PR is just dummy and overall draft. Once we have a agreement on UX design on this PR. I will create a new PR. Maybe we can treat this PR as a brainstorm thing.